app/models/comment.rb

```diff
@@ -0,0 +1,18 @@
+class Comment < ApplicationRecord
+  belongs_to :user, foreign_key: 'author_id'
+  belongs_to :post
+
+  validates :author_id, presence: true
+  validates :post_id, presence: true
+  validates :text, presence: true
+
+  after_save :update_comments
+
+  private
+
+  def update_comments
+    post = self.post
+    post.update(comments_counter: post.comments.count)
+    comments.save
+  end
```

Comment:
- [ ] Once again the `update_comments` method will be used by external entities outside the class as such I'd suggest you remove the `private` access modifier.
- [ ] [OPTIONAL] Although this is optional I'd highly recommend you rename the method to something like `update_comments_counter` as the current name is a bit misleading because you won't be updating the comments but rather the counter.

Reply:
Noted, changed to public